### ISSUE TITLE
Redis::funnel() method is undefined

### ISSUE THREAD

**Issue by mstralka** [laravel/framework#32007]
- Laravel Version: 6.18.1
- PHP Version: 7.2.28-3+ubuntu18.04.1+deb.sury.org+1
- Database Driver & Version: MySQL 5.7.29-0ubuntu0.18.04.1

### Description:
Running Laravel Homestead with Redis Server version 5.0.8 and the `predis` client, I'm trying to use the Redis::funnel() function like this:

~~~php
        Redis::funnel($key)->limit(1)->then(function () use ($elasticService, $indexingService) {
            $this->_handle($elasticService, $indexingService);
        }, function () {
            // Could not obtain a lock
            Log::debug(sprintf('Another instance of the %s job is already running for team %s. Putting this job back into the queue', self::class, $this->team->id));
            $this->release(30);
        });
~~~

But this exception is thrown:
~~~bash
[2020-03-17 14:59:54] local.ERROR: Call to undefined method Redis::funnel() {"exception":"[object] (Symfony\\Component\\Debug\\Exception\\FatalThrowableError(code: 0): Call to undefined method Redis::funnel() at /home/vagrant/sites/[project name]/app/Jobs/Search/UploadTeamSearchFile.php:61)
[stacktrace]
~~~

### Steps To Reproduce:

Try to run the Redis::funnel with Laravel 6.18.1 and Redis server 5.0.8

**Comment by driesvints** [comment]
Share UploadTeamSearchFile.php ?

**Comment by mstralka** [comment]
I actually have that code in the `handle()` method of `UploadTeamSearchFile.php`. Here is the rest of it, but with the protected `_handle()` method excluded because the contents of that function are not relevant:

~~~php
class UploadTeamSearchFile implements ShouldQueue
{
    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
    /**
     * @var Team
     */
    private $team;
    /**
     * @var bool
     */
    private $force;

    /**
     * Create a new job instance.
     *
     * @param Team $team
     * @param bool $force
     */
    public function __construct(Team $team, bool $force = false)
    {
        //
        $this->team = $team;
        $this->force = $force;
        $this->onQueue(Constants::QUEUE_SEARCH);
    }

    /**
     * Execute the job.
     *
     * @param ElasticService $elasticService
     * @param IndexingService $indexingService
     * @return void
     */
    public function handle(ElasticService $elasticService, IndexingService $indexingService)
    {
        $key = sprintf('%s-%s', self::class, $this->team->id);
        Redis::funnel($key)->limit(1)->then(function () use ($elasticService, $indexingService) {
            $this->_handle($elasticService, $indexingService);
        }, function () {
            // Could not obtain a lock
            Log::debug(sprintf('Another instance of the %s job is already running for team %s. Putting this job back into the queue', self::class, $this->team->id));
            $this->release(30);
        });
    }
   protected function _handle(ElasticService $elasticService, IndexingService $indexingService)
   {
      // not relevant
   }
}
~~~

**Comment by rs-sliske** [comment]
which redis driver are you using?

**Comment by mstralka** [comment]
predis/predis v1.1.1

**Comment by themsaid** [comment]
If you have `phpredis` installed you need to rename the Redis Facade to something else in your `app.php` config file:

```
'RedisManager' => Illuminate\Support\Facades\Redis::class,
```

**Comment by mstralka** [comment]
Hi Mohamed,

I don't have `phpredis` installed, I'm using `predis`.

Here is my partial list of alphabetical composer packages showing that `predis` is the only Redis-related package installed:

~~~bash
onelogin/php-saml                        3.0.0.x-dev f9e44f9       OneLogin PHP SAML Toolkit
opis/closure                             3.5.1                     A library that can be used to serialize closures (anonymous functions) and arbitrary objects.
paragonie/random_compat                  v9.99.99                  PHP 5.x polyfill for random_bytes() and random_int() from PHP 7
paragonie/sodium_compat                  v1.12.2                   Pure PHP implementation of libsodium; uses the PHP extension if it exists
phar-io/manifest                         1.0.3                     Component for reading phar.io manifest information from a PHP Archive (PHAR)
phar-io/version                          2.0.1                     Library for handling version information and constraints
phenx/php-font-lib                       0.5.1                     A library to read, parse, export and make subsets of different types of font files.
phenx/php-svg-lib                        v0.3.3                    A library to read, parse and export to PDF SVG files.
php-http/client-common                   2.1.0                     Common HTTP Client implementations and tools for HTTPlug
php-http/curl-client                     2.1.0                     PSR-18 and HTTPlug Async client with cURL
php-http/discovery                       1.7.4                     Finds installed HTTPlug implementations and PSR-7 message factories
php-http/guzzle6-adapter                 v2.0.1                    Guzzle 6 HTTP Adapter
php-http/httplug                         2.1.0                     HTTPlug, the HTTP client abstraction for PHP
php-http/message                         1.8.0                     HTTP Message related tools
php-http/message-factory                 v1.0.2                    Factory interfaces for PSR-7 HTTP Message
php-http/multipart-stream-builder        1.1.0                     A builder class that help you create a multipart stream
php-http/promise                         v1.0.0                    Promise used for asynchronous HTTP requests
php-school/cli-menu                      3.2.0                     A command line menu helper in PHP
php-school/terminal                      0.2.1                     A command line terminal utility in PHP
phpdocumentor/reflection-common          2.0.0                     Common reflection classes used by phpdocumentor to reflect the code structure
phpdocumentor/reflection-docblock        5.1.0                     With this component, a library can provide support for annotations via DocBlocks or otherwise retrieve inform...
phpdocumentor/type-resolver              1.1.0                     A PSR-5 based resolver of Class names, Types and Structural Element Names
phpoption/phpoption                      1.7.2                     Option Type for PHP
phpseclib/phpseclib                      2.0.25                    PHP Secure Communications Library - Pure-PHP implementations of RSA, AES, SSH2, SFTP, X.509 etc.
phpspec/prophecy                         v1.10.3                   Highly opinionated mocking framework for PHP 5.3+
phpunit/php-code-coverage                7.0.10                    Library that provides collection, processing, and rendering functionality for PHP code coverage information.
phpunit/php-file-iterator                2.0.2                     FilterIterator implementation that filters files based on a list of suffixes.
phpunit/php-text-template                1.2.1                     Simple template engine.
phpunit/php-timer                        2.1.2                     Utility class for timing
phpunit/php-token-stream                 3.1.1                     Wrapper around PHP's tokenizer extension.
phpunit/phpunit                          8.5.2                     The PHP Unit Testing framework.
piwik/piwik-php-tracker                  1.5.2                     PHP Client for Piwik Analytics Tracking API
predis/predis                            v1.1.1                    Flexible and feature-complete Redis client for PHP and HHVM
propaganistas/laravel-phone              4.2.3                     Adds phone number functionality to Laravel and Lumen based on Google's libphonenumber API.
psr/cache                                1.0.1                     Common interface for caching libraries
psr/container                            1.0.0                     Common Container Interface (PHP FIG PSR-11)
psr/http-client                          1.0.0                     Common interface for HTTP clients
psr/http-factory                         1.0.1                     Common interfaces for PSR-7 HTTP message factories
psr/http-message                         1.0.1                     Common interface for HTTP messages
psr/log                                  1.1.2                     Common interface for logging libraries
psr/simple-cache                         1.0.1                     Common interfaces for simple caching
psy/psysh                                v0.9.12                   An interactive shell for modern PHP.
pubnub/pubnub                            4.1.5                     The Only Global Network For Real-Time Data -- PubNub!
pusher/pusher-php-server                 v4.1.1                    Library for interacting with the Pusher REST API
ralouphie/getallheaders                  3.0.3                     A polyfill for getallheaders.
~~~

This is only happening in my homestead environment. Staging and production are at AWS and using AWS ElasticCache Redis version 2.8.24.

Per your suggestion, I changed `app.php` aliases to look like this:

~~~php
        'Redis' => Illuminate\Support\Facades\Redis::class,
        'RedisManager' => Illuminate\Support\Facades\Redis::class,
~~~

Then in `UploadTeamSearchFile`, I changed `Redis::funnel` to `RedisManager::funnel`. I'm still using the `Redis` facade elsewhere in the application. This was the only class using the `funnel` function.

I don't understand why, but doing that makes `UploadTeamSearchFile` work now in Homestead.

Weird.
